### PR TITLE
setup: add poetry

### PR DESCRIPTION
**Pull request by PascalEgn**
ref: https://github.com/cern-sis/issues-scoap3/issues/145

### CODE REVIEW

File: compose/local/django/Dockerfile [hunk 1]
```diff
@@ -1,55 +1,30 @@
...
-ARG BUILD_ENVIRONMENT=local
 ARG APP_HOME=/app
+WORKDIR ${APP_HOME}
 
-ENV PYTHONUNBUFFERED 1
-ENV PYTHONDONTWRITEBYTECODE 1
-ENV BUILD_ENV ${BUILD_ENVIRONMENT}
 
-WORKDIR ${APP_HOME}
+ARG BUILD_ENVIRONMENT=local
```
Comment (MJedr) [inline compose/local/django/Dockerfile]: why we need this?

File: compose/local/django/Dockerfile [hunk 2]
```diff
@@ -1,55 +1,30 @@
...
+    PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    BUILD_ENV=${BUILD_ENVIRONMENT}
 
-# Install required system dependencies
 RUN apt-get update && apt-get install --no-install-recommends -y \
-  # psycopg2 dependencies
-  libpq-dev \
-  # Translations dependencies
-  gettext \
```
Comment (MJedr) [inline compose/local/django/Dockerfile:41]: are you sure w don't need those?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+django-redis = "^5.2.0"
+djangorestframework = "^3.14.0"
+django-cors-headers = "^3.14.0"
+drf-spectacular = "^0.26.1"
+django-webpack-loader = "^1.8.1"
+gunicorn = {version = "^20.1.0", extras = ["production"]}
+psycopg2 = {version = "^2.9.6", extras = ["production"]}
+sentry-sdk = {version = "^1.19.1", extras = ["production"]}
+django-storages = {extras = ["production"], version = "^1.13.2"}
+django-anymail = {version = "^9.1", extras = ["production"]}
```
Comment (MJedr) [inline pyproject.toml]: 1. We'll have just one version of requirements (not local & production)
2. where are `[tool.poetry.extras]`?

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,70 @@
+[tool.poetry]
+name = "scoap3"
+version = "0.1.0"
+description = ""
+authors = ["CERN SIS-TS <cern-sis-ts-admins@cern.ch>"]
+license = "MIT License"
+
+[tool.poetry.dependencies]
+python = "~3.10"
```
Comment (MJedr) [inline pyproject.toml]: can't we support python 11? Did you try it?

File: compose/local/django/Dockerfile (hunk 2 above)
Comment (PascalEgn) [reply compose/local/django/Dockerfile:41]: we need those, they are within this line https://github.com/SCOAP3/scoap3/blob/84341ce9912b22012e59c754da2f4fe3fc11e118/compose/local/django/Dockerfile#L15

File: pyproject.toml (hunk 3 above)
Comment (PascalEgn) [reply pyproject.toml]: > * where are `[tool.poetry.extras]`?

What do we need them for?

File: pyproject.toml (hunk 4 above)
Comment (PascalEgn) [reply pyproject.toml]: bumped it to python 11 now, docker-compose, the application and the tests still work so I guess we are fine.

File: compose/local/django/Dockerfile (hunk 1 above)
Comment (PascalEgn) [reply compose/local/django/Dockerfile]: we don't, I removed it

File: pyproject.toml (hunk 3 above)
Comment (MJedr) [reply pyproject.toml]: it's ok now, you put `{extras = ["production"], version = "^1.13.2"}`, so then you need to have `[tool.poetry.extras]` section

File: pyproject.toml (hunk 4 above)
Comment (MJedr) [reply pyproject.toml]: that's great!

**Review by MJedr — APPROVED** [review]

**Pull request merged**